Fix protoc-gen-swagger not to output gRPC method summary and descriptions as Swagger's them

co3k opened the pull request.

achew22 commented:
@co3k, thanks so much for your pull request. Looking through, I'm a bit puzzled by what it should do. It looks like the changes to the golden `.go` files all come from changes upstream. If this code path isn't exercised by our examples, do you think you could add an example that does exercise it so I can see the new functionality?

co3k commented:
@achew22 
> It looks like the changes to the golden .go files all come from changes upstream

I added these changes to pass build on Travis CI. In the other pull request, I've encountered build error that is caused by the upstream changes. https://travis-ci.org/grpc-ecosystem/grpc-gateway/builds/390065893

> do you think you could add an example that does exercise it so I can see the new functionality?

You're right. I've added the https://github.com/grpc-ecosystem/grpc-gateway/pull/667/commits/ce1532e41996ea402ff4bd1529688490b245d1a3 commit to show it.

achew22 commented on examples/proto/examplepb/a_bit_of_everything.proto:
```diff
@@ -189,6 +189,9 @@ service ABitOfEverythingService {
 
+  // Create a new ABitOfEverything
+  //
+  // This API creates a new ABitOfEverything
```
Oops, I left that comment on the commit directly -- sorry. Can you make the indentation match the rest of the file?

co3k replied:
Oh, sorry for the mistakes. I fixed it by https://github.com/grpc-ecosystem/grpc-gateway/pull/667/commits/4b6f8895524282af806fda06e42705cbe400b817

johanbrandhorst commented:
Hi! Can you please rebase this on top of master? There was recently a change to the swagger generator. You can run something like:
```
make realclean && make examples SWAGGER_CODEGEN="java -jar $HOME/local/swagger-codegen-cli.jar"
```
from the root of the repo to regenerate all examples. This will require swagger-codegen (and Java) installed.

co3k commented:
@johanbrandhorst I rebased my changes and tests on Travis CI are passed.

johanbrandhorst commented:
Hi, could you squash the commits as well please? Sorry I didn't specify this in the last message.

co3k commented:
@johanbrandhorst OK, I've just squashed them.

johanbrandhorst commented:
Thanks, I will leave it for @achew22 to approve and merge, but I'm a little confused by the PR description. Looking at the code this change looks like it parses the method comment and adds it to "Summary" and "Description" in accordance with other such comment parsing, but the PR title, description and commit message is a little confusing. Could you reword it for clarity please?

achew22 commented:
@johanbrandhorst another neat thing about the squash button on github is that you enter the description of the new squashed commit which makes it perfect for these situations. 2 commits squashed down into 1 and then the description hopefully clarified.

@co3k Thanks so much for your contribution! This will help in producing great documentation for APIs.
